checkov/ansible/utils.py

```diff
@@ -89,3 +95,90 @@ def parse_file(
+    for name in task:
+        if name in TASK_RESERVED_KEYWORDS:
+            continue
+
+        if prefix:
+            # if the task is found in a block, then prefix the module name with 'block'
+            name = f"{prefix}{name}"
+
+        return f"{ResourceType.TASKS}.{name}.{task_name}"
+
+    return None
+
+
+def build_definitions_context(
+    definitions: dict[str, dict[str, Any] | list[dict[str, Any]]],
+    definitions_raw: dict[str, list[tuple[int, str]]],
+) -> dict[str, dict[str, Any]]:
+    definitions_context: dict[str, dict[str, Any]] = {}
+
+    for file_path, definition in definitions.items():
+        file_path_context: dict[str, Any] = {}
+        definition_raw = definitions_raw[file_path]
+
+        if not isinstance(definition, list):
+            logger.info(f"File {file_path} has the wrong type {type(definition)}")
+            continue
+
+        for code_block in definition:
+            if ResourceType.TASKS in code_block:
+                for task in code_block[ResourceType.TASKS]:
+                    _process_blocks(definition_raw=definition_raw, file_path_context=file_path_context, task=task)
+            else:
+                _process_blocks(definition_raw=definition_raw, file_path_context=file_path_context, task=code_block)
+
+        definitions_context[file_path] = file_path_context
+
+    return definitions_context
+
+
+def _process_blocks(
+    definition_raw: list[tuple[int, str]],
+    file_path_context: dict[str, Any],
+    task: Any,
+    prefix: str = "",
+) -> None:
+    """Checks for possible block usage"""
+
+    if not task or not isinstance(task, dict):
+        return
+
+    if ResourceType.BLOCK in task and isinstance(task[ResourceType.BLOCK], list):
+        prefix += f"{ResourceType.BLOCK}."  # with each nested level an extra block prefix is added
+        block_name = f"{prefix}.{task.get('name') or 'unknown'}"
+        resource_context = _create_resource_context(definition_raw=definition_raw, resource=task)
+        file_path_context[block_name] = resource_context
+
+        for block_task in task[ResourceType.BLOCK]:
+            _process_blocks(
+                definition_raw=definition_raw, file_path_context=file_path_context, task=block_task, prefix=prefix
+            )
+    else:
+        resource_context = _create_resource_context(definition_raw=definition_raw, resource=task)
+        task_name = generate_task_name(task=task, prefix=prefix)
+        if task_name:
+            file_path_context[task_name] = resource_context
+
+
+def _create_resource_context(definition_raw: list[tuple[int, str]], resource: dict[str, Any]) -> dict[str, Any]:
+    """Creates the resource context block"""
+
+    start_line = resource[START_LINE]
+    end_line = resource[END_LINE]
+    code_lines = definition_raw[start_line - 1 : end_line - 1]
+    skipped_checks = collect_suppressions_for_context(code_lines=code_lines)
+
+    return {
+        "start_line": start_line + 1,  # lines start with index 0
```

Comment:
this is a bit inconsistent for me with the code lines calculation (for example if we start with 0, can the code line start with -1?)
I can also see that in the runner it was `[start_line - 1:end_line + 1]`

Reply:
need to double check, I think you could be right, that I'm shifting the code lines a bit too often